tools/Custom/PSCmdletExtensions.cs

```diff
@@ -28,7 +30,7 @@ internal static string UnescapeString(this PSCmdlet cmdlet, string value)
             try
             {
                 var unescapedString = Uri.UnescapeDataString(value);
-                return unescapedString.Replace('+', ' ');
+                return EnclosedStringRegex.IsMatch(value) ? unescapedString: unescapedString.Replace('+', ' ');
```

Comment:
I think for this search, a regex pattern isn't really required. You can use `StartsWith` and `EndsWith` here and it should make the intent clearer. It should also perform better.

Reply:
For the given solution, an example of a value the extension method is receiving is something like ```displayName eq 'P+R | Workspace'```. So, we are only looking for this part.  ``'P+R | Workspace'`` in that string. The value could assume any pattern. e.g. ``memberOf like 'D+R TestMember'``. For such scenarios, the goal is to ensure that the plus (+) sign inside single quotes are not replaced with an empty space.

Reply:
The suggestion has been done and it's working as expected. Thanks